controller/resource_blackbox_test.go

```diff
@@ -226,3 +224,99 @@ func (rest *TestResourceREST) TestFailRegisterResourceUnknownOwner() {
 
 	test.RegisterResourceNotFound(rest.T(), rest.service.Context, rest.service, rest.securedController, payload)
 }
+
+func (rest *TestResourceREST) TestUpdateResource() {
+	// Create the resource first
+	payload := &app.RegisterResourcePayload{
+		Name:             "Resource_Alpha",
+		ParentResourceID: nil,
+		ResourceScopes:   []string{},
+		ResourceOwnerID:  rest.testIdentity.ID.String(),
+		Type:             "openshift.io/resource/area",
+	}
+
+	_, created := test.RegisterResourceCreated(rest.T(), rest.service.Context, rest.service, rest.securedController, payload)
+
+	require.NotNil(rest.T(), created)
+	require.NotNil(rest.T(), created.ID)
+
+	_, readResource := test.ReadResourceOK(rest.T(), rest.service.Context, rest.service, rest.securedController, *created.ID)
+
+	require.EqualValues(rest.T(), created.ID, readResource.ResourceID)
+	require.EqualValues(rest.T(), payload.Name, readResource.Name)
+
+	var updatedName = "Resource_Bravo"
+	updatePayload := &app.UpdateResourcePayload{
+		Name: &updatedName,
+	}
+
+	_, updated := test.UpdateResourceOK(rest.T(), rest.service.Context, rest.service, rest.securedController, *created.ID, updatePayload)
+
+	// First confirm we get the correct resource ID back in the response
+	require.EqualValues(rest.T(), created.ID, updated.ID)
+
+	// Read the resource again, and check the name has been updated
+	_, readResource = test.ReadResourceOK(rest.T(), rest.service.Context, rest.service, rest.securedController, *created.ID)
+	require.EqualValues(rest.T(), updatedName, readResource.Name)
```

Comment:
Also check if only name was updated. It will help us to make sure we don't reset the other fields to the default values if the payload doesn't contain all the fields.

Reply:
Done.